dbt/include/duckdb/macros/materializations/incremental.sql

```diff
@@ -1,6 +1,8 @@
 {% materialization incremental, adapter="duckdb", supported_languages=['sql', 'python'] -%}
 
   {%- set language = model['language'] -%}
+  -- only create temp tables if using local duckdb, as it is not currently supported for remote databases
+  {%- set temporary = not adapter.config.credentials.is_motherduck -%}
```

Comment:
ah yeah I'd like this to be an adapter method (like `adapter.is_motherduck()` ) vs. exposing the whole call chain in here; I like to abstract this stuff away from the jinja as much as I can